### PR TITLE
Feature/BC-182 Student task test case

### PR DESCRIPTION
**Pull request by Uzaeir**
# Description

  Adds the test case for new task page based on nuxt client for the student.

In this test case, student login with their credentials and see the open task in the new task menu, then student will submit the task and naviagate back to the new task menu to see the updated status of the task i.e. from open to complete task.

## Links to Tickets or other pull requests

https://ticketsystem.hpi-schul-cloud.org/browse/BC-182

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.hpi-schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Review by odalys-dataport — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -40,27 +40,28 @@
 	"dependencies": {
 		"@cucumber/cucumber": "^7.3.1",
 		"@cucumber/pretty-formatter": "^1.0.0-alpha.1",
-		"axios": "^0.21.1",
+		"axios": "^0.21.4",
```
Comment (CeEv) [inline package.json:43]: Only a hint, for package updates it is better to do it in a additional pr. You can use npm ci instant of npm i to install the packages without updates. But the updates do not show critical, also it is fine for this pr.

File: page-objects/pages/TASKListPage.js [hunk 2]
```diff
@@ -16,6 +17,9 @@ const taskTitleContainer = '.assignment.card .title';
 const taskDescriptionContainer = '.assignment .text-muted.ckcontent';
 const taskContainer = '.homework li.card';
 const deleteTaskButtonInPopup = '.delete-modal button.btn-submit';
+const clickWithoutDueDate = "//*[text()='Task19']";
+const ungradedTaskSection = "//div[@class ='v-expansion-panel']//button[@type='button']";
```
Comment (CeEv) [inline page-objects/pages/TASKListPage.js:21]: If you have time in daily i am interest what the effect of // is, to know it in future. :)

File: step_definitions/tasks/nuxtSubmitTask_steps.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    await TASKPage.clickCompletedTab();
+});
+
+When(/^.* clicks at ungraded task section$/, async function (){
+    await TASKListPage.clickUngradedTaskSection();
+});
+
+When(/^.* clicks at Task19$/, async function (){
+    await TASKListPage.clickUngradedTask();
+});
```
Comment (CeEv) [inline step_definitions/tasks/nuxtSubmitTask_steps.js:34]: linter: Missing line break after last line.

File: step_definitions/tasks/submitTask_steps.js [hunk 4]
```diff
@@ -41,10 +41,10 @@ When(/^.* clicks on Comment-Grading tab$/, async function () {
...
 //THEN
-Then(/^.*task rating is '([^']*)'%$/, async function (rating) {
+Then(/^.* task rating is '([^']*)'%$/, async function (rating) {
     await TASKPage.isTaskRating(rating);
 });
 
-Then(/^.*task remark is '([^']*)'$/, async function (remark) {
+Then(/^.* task remark is '([^']*)'$/, async function (remark) {
     await TASKPage.isTaskRemark(remark);
 });
```
Comment (CeEv) [inline step_definitions/tasks/submitTask_steps.js:50]: linter: Missing line break after last line.

**Review by CeEv — CHANGES_REQUESTED** [review]

**Review by CeEv — APPROVED** [review]

**Review by CeEv — APPROVED** [review]

**Pull request merged**